### ISSUE TITLE
How to configure the provider during cluster creation?

### ISSUE THREAD

**Issue by encron** [kubernetes-sigs/aws-encryption-provider#18]
**What would you like to be added**:

Currently the README only describes how to add the encryption provider to a running cluster by first deploying the provider and then changing the api-server flags. I was wondering how/if it is supposed to be deployed when starting a new cluster, so already starting the `apiserver` with the encryption enabled. 

For example, I am using kubeadm to provision my cluster and would add these lines to the config file:

```
  extraArgs:
    cloud-provider: aws
    encryption-provider-config: /etc/kubernetes/encryption-config.yml
  extraVolumes:
  - name: "encryption-config"
    hostPath: " /etc/kubernetes/encryption-config.yml"
    mountPath: "/etc/kubernetes/encryption-config.yml"
    readOnly: true
    pathType: File
```

This will start the apiserver with the encryption-config mount and the `--encryption-provider-config` flag and after the `apiserver` is live, I would deploy the encryption-provider.  

Is this possible or should the provider always be deployed first? From what I understand, the unix socket will need to be there before the `apiserver` can start with the `--encryption-provider-config` flag which means it won't be possible.  

**Why is this needed**:

It's not clear on how to configure the encryption-provider on a fresh cluster or if it's even possible.

**Comment by micahhausler** [comment]
@encron the encryption provider needs to be running prior to the API server in order to ensure secrets are encrypted for a brand new cluster, but it is totally possible to use the encryption provider on a new cluster. Is there anything else you need?

**Comment by rajatjindal** [comment]
Hi @micahhausler 

I am in same boat trying to setup encryption provider on a new cluster using kops. After a few hiccups I was able to get this working, but had to make a few changes for that.

1. Run aws-encryption-provider as static pod for kubernetes
2. Ensure aws-encryption-provider uses hostNetwork as CNI is not yet available
3. Ensure aws-encryption-provider health-check uses port other than 8080 as port 8080  is also used by api-server
4. Modify api-server spec (which is another static-pod) using lifecycle hook of kops
5. Ensure the master IAM role has KMS permissions

With the above changes, I was able to get the cluster up and running. 

I am willing to put a PR to add this to documentation if you think these steps looks fine to you.

Many thanks
Rajat Jindal

**Comment by micahhausler** [comment]
@rajatjindal That would be a great PR!

**Comment by rajatjindal** [comment]
this can be closed as the PR #20 has been merged now. thanks.

**Comment by encron** [comment]
Thanks for the documentation @rajatjindal , using a static pod is a good idea to circumvent this chicken-egg scenario 👍
